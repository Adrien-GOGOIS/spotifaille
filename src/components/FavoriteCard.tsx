import { FunctionComponent, useEffect, useState } from "react";
import { FavoriteTrack } from "../types/types";
import { favoriteServiceInstance } from "../services/favorite.service";
import FavoriteActionButton from "./FavoriteActionButton";

interface Props {
	favorite: FavoriteTrack;
}

const FavoriteCard: FunctionComponent<Props> = ({favorite}: Props) => {
	const [isActive, setIsActive] = useState<boolean>(true);

	return (
		<div className="relative w-1/2 max-w-sm rounded overflow-hidden shadow-lg h-auto">
			{/* {!isActive && <p className="absolute top-28 left-20 text-red-500 text-xl z-10 bg-white p-4 rounded border-2 border-red-500">REMOVED !</p>} */}
			<img className={isActive ? "w-full" : "blur-sm w-full"} src={favorite.track.album.images[0].url} alt="Sunset in the mountains"/>
			<div className="px-6 py-4">
				<div className="font-bold text-xl mb-2">{favorite.track.artists[0].name}</div>
				<div className="font-bold text-lg mb-2">{favorite.track.album.name}</div>
				<p className="text-gray-700 text-base">{favorite.track.name}</p>
			</div>
			<div className="mt-16">
				<FavoriteActionButton isActive={isActive} setIsActive={setIsActive} favorite={favorite}/>
			</div>
		</div>
	)
}

export default FavoriteCard;