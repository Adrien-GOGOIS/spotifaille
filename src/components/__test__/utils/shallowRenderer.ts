/* eslint-disable @typescript-eslint/no-magic-numbers */
/* eslint-disable @typescript-eslint/no-type-alias */
import ShallowRenderer from 'react-test-renderer/shallow';

const renderer = ShallowRenderer.createRenderer();

type Input = Parameters<typeof renderer.render>[0];
type Output = ReturnType<typeof renderer.getRenderOutput>;

export const shallowRender = (node: Input): Output => {
  renderer.render(node);
  return renderer.getRenderOutput();
};
