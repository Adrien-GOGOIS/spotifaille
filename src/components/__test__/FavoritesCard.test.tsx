import FavoriteCard from '../FavoriteCard';
import { favoriteDataset } from './utils/dataset';

import { shallowRender } from './utils/shallowRenderer';

describe('FavoriteCard Component', () => {
  describe('Rendering', () => {
    test('should display component', () => {
      // eslint-disable-next-line testing-library/render-result-naming-convention
      const favoriteCard = shallowRender(<FavoriteCard favorite={favoriteDataset} />);

      expect(favoriteCard).toMatchSnapshot();
    });
  });
});