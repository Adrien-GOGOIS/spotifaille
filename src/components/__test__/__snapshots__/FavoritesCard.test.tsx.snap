// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteCard Component Rendering should display component 1`] = `
<div
  className="my-1 px-1 w-full md:w-1/2 lg:my-4 lg:px-4 lg:w-1/3 xl:w-1/5 rounded overflow-hidden shadow-lg h-auto relative"
>
  <img
    alt="Sunset in the mountains"
    className="w-full"
    src="[[URL]]"
  />
  <div
    className="px-6 py-4"
  >
    <div
      className="font-bold text-xl mb-2"
    >
      [[NAME]]
    </div>
    <div
      className="font-bold text-lg mb-2"
    >
      [[NAME]]
    </div>
    <p
      className="text-gray-700 text-base"
    >
      [[]]
    </p>
  </div>
  <div
    className="mt-16"
  >
    <FavoriteActionButton
      favorite={
        Object {
          "added_at": "[[DATE]]",
          "track": Object {
            "album": Object {
              "album_type": "[[ALBUM_TYPE]]",
              "artists": Array [
                Object {
                  "external_urls": Object {
                    "spotify": "[[SPOTIFY]]",
                  },
                  "href": "[[HREF]]",
                  "id": "[[ID]]",
                  "name": "[[NAME]]",
                  "type": "[[TYPE]]",
                  "uri": "[[URI]]",
                },
              ],
              "available_markets": Array [
                "[[AVAILABLE_MARKET]]",
              ],
              "external_urls": Object {
                "spotify": "[[SPOTIFY]]",
              },
              "href": "[[HREF]]",
              "id": "[[ID]]",
              "images": Array [
                Object {
                  "height": 2,
                  "url": "[[URL]]",
                  "width": 2,
                },
                Object {
                  "height": 2,
                  "url": "[[URL]]",
                  "width": 2,
                },
                Object {
                  "height": 2,
                  "url": "[[URL]]",
                  "width": 2,
                },
              ],
              "name": "[[NAME]]",
              "release_date": "[[RELEASE_DATE]]",
              "release_date_precision": "[[RELEASE_DATE_PRECISION]]",
              "total_tracks": 2,
              "type": "[[TYPE]]",
              "uri": "[[URI]]",
            },
            "artists": Array [
              Object {
                "external_urls": Object {
                  "spotify": "[[SPOTIFY]]",
                },
                "href": "[[HREF]]",
                "id": "[[ID]]",
                "name": "[[NAME]]",
                "type": "[[TYPE]]",
                "uri": "[[URI]]",
              },
            ],
            "available_markets": Array [
              "[[AVAILABLE_MARKET]]",
            ],
            "disc_number": 2,
            "duration_ms": 2,
            "explicit": true,
            "external_ids": Object {
              "isrc": "[[ISRC]]",
            },
            "external_urls": Object {
              "spotify": "[[SPOTIFY]]",
            },
            "href": "[[HREF]]",
            "id": "[[ID]]",
            "is_local": false,
            "name": "[[]]",
            "popularity": 2,
            "preview_url": "[[PREVIEW_URL]]",
            "track_number": 2,
            "type": "[[TYPE]]",
            "uri": "[[URI]]",
          },
        }
      }
      isActive={true}
      setIsActive={[Function]}
    />
  </div>
</div>
`;
